### PR TITLE
batches: remove site admin restriction on license info in JS context

### PR DESCRIPTION
**Pull request by courier-new**
Silly me committed the cardinal sin of not testing changes as a non-admin user once again. 🙃 When will I learn, eh?

`window.context.licenseInfo` was `null` for non-admins, which unintentionally locked them out of BC since it looks like the instance is running in OSS. 😬 None of the information contained in license info is sensitive, and besides the BC stuff, it's only surfaced in the UI from site admin settings anyway, so I don't think there's any risk in removing this. However, I've tagged @pjlast just in case. If at some point we add something sensitive in here, we could put the site admin guard back but just for the sensitive fields.

## Test plan

- Run the instance as enterprise and sign in as a non-admin
- Verify `window.context.licenseInfo` is not null
- Verify BC nav menu option and pages are visible

### CODE REVIEW

**Comment by eseliger** [comment]
should we ping security here, too?

**Comment by courier-new** [comment]
Sure! @sourcegraph/security-code-review FYI what all is included in here is the following:
```go
type LicenseInfo struct {
	CurrentPlan            string // e.g. "enterprise-0" or "team-0"
	CodeScaleLimit         string // static value "100GiB"
	CodeScaleCloseToLimit  bool // true if repo usage stats exceeds 90% of the above limit
	CodeScaleExceededLimit bool // true if repo usage stats exceeds the above limit
	KnownLicenseTags       []string // only present on dotcom. enumerates all the available plans by name e.g. "plan:enterprise-0" and all available features by name e.g. "code-insights"
	BatchChanges           *FeatureBatchChanges // indicates the max number changesets permitted per batch change on the given license, or if that license is unrestricted
}
```

The contents of the `licenseInfo` property on the context are null on OSS and "free" plans.

References:
- [All plans](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/licensing/data.go?L30)
- [All features](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/licensing/data.go?L45)

If any of these properties feel sensitive I can put the site admin guard back just for them. 🙂

**Review by evict — APPROVED** [review]

File: enterprise/cmd/frontend/internal/licensing/init/init.go [hunk 1]
```diff
@@ -49,11 +49,7 @@ func Init(
 	database.BeforeCreateExternalService = enforcement.NewBeforeCreateExternalServiceHook()
 
 	logger := log.Scoped("licensing", "licensing enforcement")
-	hooks.GetLicenseInfo = func(isSiteAdmin bool) *hooks.LicenseInfo {
-		if !isSiteAdmin {
-			return nil
-		}
-
+	hooks.GetLicenseInfo = func() *hooks.LicenseInfo {
```
Comment (BolajiOlajide) [inline enterprise/cmd/frontend/internal/licensing/init/init.go:52]: Could we add a comment here for other folks to know about this change?

**Review by BolajiOlajide — APPROVED** [review]

File: enterprise/cmd/frontend/internal/licensing/init/init.go (hunk 1 above)
Comment (courier-new) [reply enterprise/cmd/frontend/internal/licensing/init/init.go:56]: I did you one better and [added 3](https://github.com/sourcegraph/sourcegraph/pull/52673/commits/7675f679ebd3a1bc0bc8f42624889a9aa1086ebe). 🙂

Comment (BolajiOlajide) [reply enterprise/cmd/frontend/internal/licensing/init/init.go:56]: 3 🌮 s for you.

**Pull request merged**